packages/adapter-flagsmith/src/index.ts

```diff
@@ -0,0 +1,54 @@
+import type { Adapter } from 'flags';
+import flagsmith, { IFlagsmithFeature, IInitConfig } from 'flagsmith';
+
+type AdapterParams = {
+  key?: string;
+} & IInitConfig;
+
+export function createFlagsmithAdapter<
+  ValueType extends IFlagsmithFeature,
+  EntitiesType,
+>({
+  key: customKey,
+  ...configParams
+}: AdapterParams): Adapter<ValueType, EntitiesType> {
+  async function initialize(config: IInitConfig) {
+    if (flagsmith?.initialised) return;
+    await flagsmith.init({ fetch: global.fetch, ...configParams });
+  }
+
+  return {
+    async decide({
+      key,
+      entities,
+      headers,
+      cookies,
+      defaultValue,
+    }): Promise<ValueType> {
+      await initialize(configParams);
+      const state = flagsmith.getState();
```

Comment:
Note that if there are multiple flags, `getState()` might be called many times in one request:

```typescript
a = flag({ key: 'a', adapter: flagSmith.getFeature() })
b = flag({ key: 'a', adapter: flagSmith.getFeature() })
...
await Promise.all([a(), b()])
```

You may consider ways to reduce duplicate work